File: app/src/main/java/com/education4all/decimals/WebActivity.java
package com.education4all.decimals;

import android.annotation.TargetApi;
import android.os.Build;
import android.os.Bundle;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

public class WebActivity extends AppCompatActivity {
    private WebView webView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.web);
        Toolbar myToolbar = (Toolbar) findViewById(R.id.my_toolbar);
        setSupportActionBar(myToolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        webView = findViewById(R.id.webView);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new MyWebViewClient());
        webView.loadUrl("http://math-trainer.ru");
    }

    @Override
    public void onBackPressed() {
        if(webView.canGoBack()) {
            webView.goBack();
        } else {
            super.onBackPressed();
        }
    }

    private class MyWebViewClient extends WebViewClient {
        @TargetApi(Build.VERSION_CODES.N)
        @Override
        public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
            view.loadUrl(request.getUrl().toString());
            return true;
        }

        // Для старых устройств
        @Override
        public boolean shouldOverrideUrlLoading(WebView view, String url) {
            view.loadUrl(url);
            return true;
        }
    }

}

File: app/src/main/java/com/education4all/activities/WebActivity.kt
package com.education4all.activities

import android.annotation.TargetApi
import android.net.ConnectivityManager
import android.os.Build
import android.os.Bundle
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.education4all.R

class WebActivity : AppCompatActivity() {
    private var webView: WebView? = null
    private var connected = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        connected = CheckForConnection()
        if (connected) {
            setContentView(R.layout.web)
            webView = findViewById(R.id.webView)
            webView.getSettings().javaScriptEnabled = true
            webView.setWebViewClient(MyWebViewClient())
            webView.loadUrl("http://math-trainer.ru")
        } else {
            setContentView(R.layout.no_internet)
            //            TextView messagetext = findViewById(R.id.message);
//            messagetext.setText(message);
        }
        val myToolbar = findViewById<Toolbar>(R.id.my_toolbar)
        setSupportActionBar(myToolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
    }

    private fun CheckForConnection(): Boolean {
        var internetConnection = false
        val connectivityManager = getSystemService(CONNECTIVITY_SERVICE) as ConnectivityManager
        //        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
//            message = "Build version > M";
//            Network nw = connectivityManager.getActiveNetwork();
//            if (nw == null) {
//                internetConnection = false;
//                message += ", nw==null";
//            } else {
//                NetworkCapabilities actNw = connectivityManager.getNetworkCapabilities(nw);
//                internetConnection = actNw != null && (
//                        actNw.hasCapability(NetworkCapabilities.NET_CAPABILITY_INTERNET) ||
//                        actNw.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ||
//                        actNw.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) ||
//                        actNw.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)// ||
//                    //    actNw.hasTransport(NetworkCapabilities.TRANSPORT_BLUETOOTH)
//                );
//
//                message += actNw == null ? ", actNw==null" : ", no transport";
//            }
        //      } else {
        var message = "Build version < M"
        val nwInfo = connectivityManager.activeNetworkInfo
        internetConnection = nwInfo != null && nwInfo.isConnected
        message += if (nwInfo == null) ", nwInfo==null" else ", nw is not connected"
        //        }
        return internetConnection
    }

    override fun onBackPressed() {
        if (connected && webView!!.canGoBack()) {
            webView!!.goBack()
        } else {
            super.onBackPressed()
        }
    }

    private class MyWebViewClient : WebViewClient() {
        @TargetApi(Build.VERSION_CODES.N)
        override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
            view.loadUrl(request.url.toString())
            return true
        }

        // Для старых устройств
        override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
            view.loadUrl(url)
            return true
        }
    }
}